pkg/services/network/network_test.go

```diff
@@ -470,7 +500,124 @@ var _ = Describe("Network provider", func() {
 				err = np.VerifyNetworkStatus(ctx, clusterCtx, vnetObj)
 
 				expectedErrorMessage := fmt.Sprintf("virtual network ready status is: '%s' in cluster %s. reason: %s, message: %s",
-					"False", apitypes.NamespacedName{Namespace: dummyNs, Name: dummyCluster}, testVnetNotRealizedReason, testVnetNotRealizedMessage)
+					"False", apitypes.NamespacedName{Namespace: dummyNs, Name: dummyCluster}, testNetworkNotRealizedReason, testNetworkNotRealizedMessage)
+				Expect(err).To(MatchError(expectedErrorMessage))
+				Expect(conditions.IsFalse(clusterCtx.VSphereCluster, vmwarev1.ClusterNetworkReadyCondition)).To(BeTrue())
+			})
+		})
+
+		Context("with nsx-vpc network provider and subnetset exists", func() {
+			BeforeEach(func() {
+				client = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(configmapObj, systemNamespaceObj).Build()
+				vpcNp, _ = NSXTVpcNetworkProvider(client).(*nsxtVPCNetworkProvider)
+				np = vpcNp
+				err = np.ProvisionClusterNetwork(ctx, clusterCtx)
+			})
+
+			It("should not update subnetset", func() {
+				var initialSubnetSetSpec nsxopv1.SubnetSetSpec
+
+				// Fetch the SubnetSet before the operation
+				initialSubnetSet := &nsxopv1.SubnetSet{}
+				err = client.Get(ctx, apitypes.NamespacedName{
+					Name:      dummyCluster,
+					Namespace: dummyNs,
+				}, initialSubnetSet)
+				Expect(err).NotTo(HaveOccurred())
+				initialSubnetSetSpec = initialSubnetSet.Spec
+
+				// Presumably there's code here that might modify the SubnetSet...
+				Expect(err).ToNot(HaveOccurred())
+				subnetset, localerr := np.GetClusterNetworkName(ctx, clusterCtx)
```

Comment:
Why not just call this err?

Reply:
Same in other cases below

Reply:
I see, this is used in other cases above as well. But I don't get the point why we:
a) don't check errors that occur in BeforeEach directly in BeforeEach
b) why we store errors in localerr. It might make sense if we wanted to preserve the value of err, but we usually just overwrite it a bit below

Let's please get rid of all these localerr's and let's move the err checks up.

(fine to do it for existing code in a follow-up PR, but let's adjust the new code we add here in this PR)